sklearn/preprocessing/_polynomial.py

```diff
@@ -384,6 +390,11 @@ def transform(self, X):
         n_samples, n_features = X.shape
 
         if sparse.isspmatrix_csr(X):
+            # We can determine a priori what the final dtype for the post-stack
+            # matrix must be, and can then use this in construction the pre-
+            # stack matrices.
+            max_int32 = np.iinfo(np.int32).max
+            index_t = np.int64 if self.n_output_features_ > max_int32 else np.int32
```

Comment:
Please rename `index_t` to `index_dtype` as this is a dtype object rather than a simple C type.